consensus/consortium/v2/finality/consortium_header.go

```diff
@@ -248,6 +258,118 @@ func DecodeExtra(rawBytes []byte, isShillin bool) (*HeaderExtraData, error) {
 	return &extraData, nil
 }
 
+// extraDataRLP excludes the vanity, hasFinalityVotes
+// as they're no longer in use. Whereas, seal is not
+// rlp-encoded but appended manually.
+type extraDataRLP struct {
+	FinalityVotedValidators FinalityVoteBitSet
+	AggregatedFinalityVotes []byte
+	CheckpointValidators    []validatorWithBlsPubRLP
+}
+
+type validatorWithBlsPubRLP struct {
+	Address      common.Address
+	BlsPublicKey []byte
+}
+
+// EncodeRLP computes rlp-based encoding for HeaderExtraData before
+// appending Seal at the end of the return, enabling encodeSigHeader
+// easily to exclude Seal before signing.
+func (extraData *HeaderExtraData) EncodeRLP() ([]byte, error) {
+	var ext = &extraDataRLP{}
+	if extraData.HasFinalityVote != 0 && extraData.HasFinalityVote != 1 {
+		return nil, ErrInvalidHasFinalityVote
+	}
+	if extraData.HasFinalityVote == 1 {
+		if extraData.AggregatedFinalityVotes == nil || len(extraData.FinalityVotedValidators.Indices()) == 0 {
+			return nil, ErrInvalidExtraData
+		}
+		ext.FinalityVotedValidators = extraData.FinalityVotedValidators
+		ext.AggregatedFinalityVotes = extraData.AggregatedFinalityVotes.Marshal()
+	}
+
+	cp := make([]validatorWithBlsPubRLP, 0)
```

Comment:
minor: As we already know the length of this slice (len(extraData.CheckpointValidators)), we can use `make([]validatorWithBlsPubRLP, 0, len(extraData.CheckpointValidators))`. I wrote a simple benchmark and found that it helps to make code run faster.